src/cascadia/TerminalSettingsModel/CascadiaSettings.cpp

```diff
@@ -66,6 +66,29 @@ CascadiaSettings::CascadiaSettings(winrt::hstring json) :
     _ValidateSettings();
 }
 
+winrt::Microsoft::Terminal::Settings::Model::CascadiaSettings CascadiaSettings::Copy() const
+{
+    // dynamic profile generators added by default

```

Comment:
I'm not sure what this has to do with the dynamic profile generators.... Could you explain this comment a little more?

Reply:
Sure. So, `Copy()` is intended to be a 1-to-1 copy of another `CascadiaSettings`. That means all member variables need to be copied over too. Rather than copying `_profileGenerators` by adding `Copy()` to `IDynamicProfileGenerator`, `CascadiaSettings`'s default constructor already populates `_profileGenerators` by default. So there's no need to manually copy them. Let's just construct them again.